### PR TITLE
feat: [SIW-956] Add key attestation validation on iOS

### PR DESCRIPTION
**Pull request by grausof**
Adds key attestation on **iOS**, in particular it introduces `assertion` and `attestation` validation.

#### List of Changes
- Add `attestation-service` in infra
- Add `valiateiOSAssertion` and `valiateiOSAttestation`
- Add unit test with real attestation/assertion
- Add `AttestationServiceConfiguration`

#### How Has This Been Tested?
Unit tested

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    pipe(
+      E.tryCatch(
+        () => Buffer.from(attestation, "base64"),
+        () => new Error(`Invalid attestation: ${attestation}`)
+      ),
+      TE.fromEither,
+      TE.chainW((data) =>
+        pipe(
+          [
+            valiateiOSAttestation(
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts:38]: typo? `validate` instead?

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            () => new Error(`Invalid assertion: ${assertion}`)
+          ),
+          TE.fromEither
+        ),
+        // TODO: [SIW-969] Add getting the public hardware key from the DB with keyTag and the signCount
+        hardwareKey: TE.left(new Error("Not implemented")),
+        signCount: TE.left(new Error("Not implemented")),
+      }),
+      TE.chain(({ assertionData, hardwareKey, signCount }) =>
+        valiateiOSAssertion(
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts:80]: as above

**Review by hevelius — CHANGES_REQUESTED** [review]

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import * as T from "fp-ts/Task";
+import { sequenceS } from "fp-ts/lib/Apply";
+
+import {
+  AttestationService,
+  ValidatedAttestation,
+} from "../../attestation-service";
+import { AttestationServiceConfiguration } from "../../app/config";
+import { validateAndroidAttestation } from "./android";
+import { valiateiOSAssertion, valiateiOSAttestation } from "./ios";
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts:15]: `validate`?

**Review by hevelius — CHANGES_REQUESTED** [review]

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts (hunk 1 above)
Comment (grausof) [reply apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts]: typo, thanks!

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts (hunk 2 above)
Comment (grausof) [reply apps/io-func-wallet-solution-backend/src/infra/attestation-service/index.ts]: typo, thanks!

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/attestation.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  keyId: string;
+  bundleIdentifier: string;
+  teamIdentifier: string;
+  allowDevelopmentEnvironment: boolean;
+  appleRootCertificate: string;
+};
+
+/* This function does not lend itself well to functional programming,
+ * therefore the validation was written in imperative language
+ */
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/attestation.ts:26]: This is a personal thought on the implementation. What do you think to remove it or replace it with typical comment
  
```
/**
* function description
* @param {string} param - param description
*/
```

In general, what do you think about using a multi-line comment for the functions header description as a best practice?

**Review by hevelius — CHANGES_REQUESTED** [review]

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/assertion.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    payload,
+    hardwareKey,
+    bundleIdentifier,
+    teamIdentifier,
+    signCount,
+  } = params;
+
+  const { signature, authenticatorData } = decodedAssertion;
+
+  // Convert JWK to PEM
```
[the remark this answers is not in the document]
Comment (grausof) [reply apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/assertion.ts:29]: No, this is not step 1. Step 1 is the decode of the assertion which is done upstream. Do you think it is useful to specify it?

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/attestation.ts (hunk 4 above)
Comment (grausof) [reply apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/attestation.ts:26]: yes, agreed. I'll try to do a small refactor with a dedicated PR. I'll delete the comment for now

File: apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/assertion.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (grausof) [reply apps/io-func-wallet-solution-backend/src/infra/attestation-service/ios/assertion.ts:27]: I added a comment to clarify it better

**Pull request merged**